src/actions/index.ts

```diff
@@ -181,11 +187,21 @@ export const submitFormError : ActionCreator<ISubmitFormError> = (status : strin
   };
 }
 
-export const validateEmail = (newValue: IErrorableInput) => {
-  const invalid = !newValue.value.match(EMAIL_REGEX);
+export const validateByPattern = (newValue: IErrorableInput, pattern: RegExp, failMsg: string) => {
+  const invalid = (newValue.value == null) || !newValue.value.match(pattern);
   if (invalid) {
-    newValue.validation = "Must be a valid email address."
+    newValue.validation = failMsg;
   }
+}
+
+export const validateEmail = (newValue: IErrorableInput) => {
+  validateByPattern(newValue, EMAIL_REGEX, "Must be a valid email address.");
+  return newValue;
+}
+
+export const validateOAuthRedirectURI = (newValue: IErrorableInput) => {
+  const partialUrlPattern = /^http[s]?:[/][/][^/:?]+(:[0-9]+)?([/][^?#]*)?$/;
```

Comment:
I think this is allowing URLs with fragments (like `?test=foo`), which okra will reject. 

```suggestion
try {
  const uri = new URL(newValue.value);
  if (uri.search !== "") {
    newValue.validation = "Cannot include a query fragment";
  }
} catch (error) {
  // This happens when the URL is malformed, for instance does not include a protocol
  newValue.validation = "Must be an http or https URL.";
}
```

Reply:
The regex rejects query strings. It does accept fragments. I'm not tied to that regex but the URL class doesn't provide direct access to the fragment.

Reply:
Hmm I tried it on the linked preview paged and seemed to be able to submit with a query string.

Reply:
Ah got it the issue was elsewhere.